main.go

```diff
@@ -123,3 +129,43 @@ func main() {
 	}
 	log.Fatalln(gwServer.ListenAndServeTLS("", ""))
 }
+
+func runClient() {
+	addr := fmt.Sprintf("localhost:%d", *gRPCPort)
+	// Set up a connection to the server.
+	conn, err := grpc.Dial(addr,
+		grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(insecure.CertPool, "")),
+	)
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer conn.Close()
+	c := pbExample.NewUserServiceClient(conn)
+
+	{
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+
+		user := pbExample.User{ID: 42, Role: pbExample.Role_ADMIN}
+		_, err = c.AddUser(ctx, &user)
+		if err != nil {
+			log.Fatalf("could not create user: %v", err)
+		}
+		log.Infof("created user %+v", user)
+	}
+
+	{
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+
+		r, err := c.ListUsers(ctx, nil)
+		if err != nil {
+			log.Fatalf("could not list users: %v", err)
+		}
+		rcv, err := r.Recv()
+		if err != nil {
+			log.Fatalf("received error: %v", err)
+		}
+		log.Infof("list users: %+v", rcv)
```

Comment:
This should really loop over `r.Recv()` until `io.EOF`, not assume the size of the return, since this is supposed to be an example of how to use it.